core/src/main/java/org/infinispan/remoting/MIMECacheEntry.java

```diff
@@ -47,4 +55,35 @@ public boolean equals(Object o) {
    public int hashCode() {
       return 31 * (contentType != null ? contentType.hashCode() : 0) + (data != null ? Arrays.hashCode(data) : 0);
    }
+
+   public static class Externalizer extends AbstractExternalizer<MIMECacheEntry> {
+
+      @Override
+      public Set<Class<? extends MIMECacheEntry>> getTypeClasses() {
+         return Util.<Class<? extends MIMECacheEntry>>asSet(MIMECacheEntry.class);
+      }
+
+      @Override
+      public void writeObject(ObjectOutput out, MIMECacheEntry obj) throws IOException {
+         out.writeUTF(obj.contentType);
+         out.writeInt(obj.data.length);
+         out.write(obj.data);
```

Comment:
I think calling `obj.write(byte[])` in this case is just fine. When the new marshaller is in place (PR 2/3), we can discuss further how to come up with a single method to write byte arrays. At the end of the day, `obj.write(byte[])` should call up to that agreed implementation, whether that is `MarshallUtil.marshallByteArray` or any other impl.